### ISSUE TITLE
Firebase Performance makes Ktor requests crash

### ISSUE THREAD

**Issue by leinardi** [firebase/firebase-ios-sdk#7277]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: Version 12.2 (12B45b)
  * Firebase SDK version: 7.3.0 & 6.32.2
  * Installation method: Zip file
  * Firebase Component: Performance

### [REQUIRED] Step 2: Describe the problem
Using Ktor together with Firebase Performance make the app crash with `EXC_BAD_ACCESS`.

The only workaround seems to remove Firebase Performance.

#### Steps to reproduce:
We are able to reproduce the issue using kotlin 1.4.21, ktor 1.5.0' and coroutines 1.4.1-native-mt.

Executing a Ktor request on a project where Firebase Performance is added and will crash the app as soon as the request starts.

The Ktor request can be as simple as this:
```
suspend fun ktorRequest(): String = httpClient.get<String>("https://en.wikipedia.org/wiki/Main_Page")
```

This is the [stacktrace.txt](https://github.com/firebase/firebase-ios-sdk/files/5788425/stacktrace.txt)

#### Relevant Code:
A sample KMM client crashing with Firebase Performance here (you might need to update the dependencies to the latest version): https://github.com/ktorio/ktor/issues/1040

**Comment by maksymmalyhin** [comment]
Potentially related #6938.

**Comment by maksymmalyhin** [comment]
@leinardi Thank you reporting the issue. Could you please attach the crash log to understand if it is the same issue as described at #6938.

**Comment by leinardi** [comment]
Hi @maksymmalyhin, thank you for the quick answer. What exactly do you mean by "crash log"? I already attached the stacktrace to the OP, isn't that the crash log?

![image](https://user-images.githubusercontent.com/273338/104049752-c20e2680-51e5-11eb-8e83-6f7bcd66e9ce.png)

**Comment by maksymmalyhin** [comment]
Ah, sorry, I missed the link. 

As far as I understand, the file contains Kotlin stack trace for the crashed thread, is it correct? Would you be able to provide stack traces for other threads? Are native (iOS) thread stack traces also available?

**Comment by leinardi** [comment]
I have to check with my iOS colleague @kchromik on Monday, I am the KMM developer and he's doing the iOS integration.
We will report back on Monday, thanks for looking at the issue :+1:

**Comment by kchromik** [comment]
The native stack trace doesn't differ from the one provided by @leinardi.

**Comment by maksymmalyhin** [comment]
@leinardi @kchromik We will try to reproduce the issue on our end to proceed with the investigation.

In the meanwhile you may try to disable Firebase Performance instrumentation by setting [Performance .isInstrumentationEnabled](https://firebase.google.com/docs/reference/swift/firebaseperformance/api/reference/Classes/Performance?authuser=0#isinstrumentationenabled) property to `false` as a workaround. With `isInstrumentationEnabled = false` Performance SDK won't do any runtime modifications such ISA swizzling some objects to get network request stats, etc.

**Comment by kchromik** [comment]
Thanks @maksymmalyhin that fixed the issue.

**Comment by leinardi** [comment]
> Thanks @maksymmalyhin that fixed the issue.

Let's say that worked around the issue 😅

A proper fix is still needed 😉 (please do not close this issue!)

**Comment by raymondlam** [comment]
An update, I was able to reproduce the crash using https://github.com/kotlin-hands-on/kmm-networking-and-data-storage as a sample app with the Firebase performance SDK and the current release for Kotlin.

I have confirmed that the crash is fixed once I switched the app to use the [Kotlin EAP preview v1.4.30-RC](https://github.com/JetBrains/kotlin/releases/tag/v1.4.30-RC) which contains the fix in https://github.com/JetBrains/kotlin-native/pull/4569!

I'll keep this issue open until the next release of Kotlin is publicly available.
